Flaky-test: PulsarFunctionLocalRunTest.testE2EPulsarFunctionLocalRunMultipleInstances

PulsarFunctionLocalRunTest. testE2EPulsarFunctionLocalRunMultipleInstances is flaky. It fails sporadically.

[example failure](https://github.com/apache/pulsar/runs/7477945933?check_suite_focus=true)

```
  Error:  Tests run: 39, Failures: 1, Errors: 0, Skipped: 35, Time elapsed: 49.925 s <<< FAILURE! - in org.apache.pulsar.functions.worker.PulsarFunctionLocalRunTest
  Error:  testE2EPulsarFunctionLocalRunMultipleInstances(org.apache.pulsar.functions.worker.PulsarFunctionLocalRunTest)  Time elapsed: 3.108 s  <<< FAILURE!
  java.lang.AssertionError: expected [my-message-1!] but found [my-message-4!]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
  	at org.testng.Assert.assertEquals(Assert.java:122)
  	at org.testng.Assert.assertEquals(Assert.java:629)
  	at org.testng.Assert.assertEquals(Assert.java:639)
  	at org.apache.pulsar.functions.worker.PulsarFunctionLocalRunTest.testE2EPulsarFunctionLocalRun(PulsarFunctionLocalRunTest.java:509)
  	at org.apache.pulsar.functions.worker.PulsarFunctionLocalRunTest.lambda$testE2EPulsarFunctionLocalRunMultipleInstances$11(PulsarFunctionLocalRunTest.java:750)
  	at org.apache.pulsar.functions.worker.PulsarFunctionLocalRunTest.runWithPulsarFunctionsClassLoader(PulsarFunctionLocalRunTest.java:1182)
  	at org.apache.pulsar.functions.worker.PulsarFunctionLocalRunTest.testE2EPulsarFunctionLocalRunMultipleInstances(PulsarFunctionLocalRunTest.java:750)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

<!-- optionally provide the full stacktrace ->
<details>
<summary>Full exception stacktrace</summary>
<code><pre>
full exception stacktrace here
</pre></code>
</details>

--- a/pulsar-broker/src/test/java/org/apache/pulsar/functions/worker/PulsarFunctionLocalRunTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/functions/worker/PulsarFunctionLocalRunTest.java
@@ -27,7 +27,6 @@ import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertNotEquals;
 import static org.testng.Assert.assertTrue;
 import static org.testng.Assert.fail;
-
 import com.google.common.collect.Lists;
 import com.google.common.collect.Sets;
 import java.io.File;
@@ -47,7 +46,6 @@ import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
-
 import lombok.Cleanup;
 import org.apache.pulsar.broker.PulsarService;
 import org.apache.pulsar.broker.ServiceConfiguration;
@@ -449,6 +447,7 @@ public class PulsarFunctionLocalRunTest {
 
         functionConfig.setJar(jarFilePathUrl);
         functionConfig.setParallelism(parallelism);
+        functionConfig.setRetainOrdering(true);
         int metricsPort = FunctionCommon.findAvailablePort();
         @Cleanup
         LocalRunner localRunner = LocalRunner.builder()

